src/xmipp/libraries/reconstruction/volume_subtraction.cpp

```diff
@@ -52,6 +52,35 @@ void POCSFourierAmplitude(const MultidimArray<double> &A, MultidimArray< std::co
 		}
 }
 
+void POCSFourierAmplitudeRadAvg(MultidimArray< std::complex<double> > &V, double lambda, MultidimArray<double> &rQ, int V1size_x, int V1size_y, int V1size_z)
+{
+	int V1size2_x = V1size_x/2;
+	double V1sizei_x = 1.0/V1size_x;
+	int V1size2_y = V1size_y/2;
+	double V1sizei_y = 1.0/V1size_y;
+	int V1size2_z = V1size_z/2;
+	double V1sizei_z = 1.0/V1size_z;
+	double wx, wy, wz;
+
+	for (int k=0; k<V1size_z; ++k)
+		{
+			FFT_IDX2DIGFREQ_FAST(k,V1size_z,V1size2_z,V1sizei_z,wz);
+			double wz2 = wz*wz;
+			for (int i=0; i<V1size_y; ++i)
+			{
+				FFT_IDX2DIGFREQ_FAST(i,V1size_y,V1size2_y,V1sizei_y,wy);
+				double wy2 = wy*wy;
+				for (int j=0; j<V1size_x; ++j)
+				{
+					FFT_IDX2DIGFREQ_FAST(j,V1size_x,V1size2_x,V1sizei_x,wx);
+					double w = sqrt(wx*wx + wy2 + wz2);
+					int iw = (int)round(w*V1size_x); // size_x??
+					DIRECT_A3D_ELEM(V,k,i,j)*=(1-lambda);//+lambda*DIRECT_MULTIDIM_ELEM(rQ,iw);
```

Comment:
unless there's some side effect I'm not aware of, this whole block does not do anything else but multiplies some values of `V` by constant `1-lambda`.